### PR TITLE
docs: moving everything around

### PR DESCRIPTION
**Pull request by philsturgeon**
## Checklist

- [ ] Tests have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## What kind of change does this PR introduce?

Docs update. Basically adding validation proxy guides with some stuff from the blog but also shuffling everything around to make it fit better.

### CODE REVIEW

File: .github/workflows/automerge.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  status: {}
+jobs:
+  automerge:
+    runs-on: ubuntu-latest
+    steps:
+      - name: automerge
+        if: github.actor != 'dependabot-preview[bot]' && github.actor != 'dependabot[bot]'
+        uses: "pascalgn/automerge-action@v0.6.0"
+        env:
+          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
```
Comment (XVincentX) [inline .github/workflows/automerge.yml]: 1. Why did you change this workflow?
2. dependant is already able to auto merge PRs that are approved and I've even configured the autorebase so all the updates merge themselves as long the test pass. Did you want to change something in particular here?

File: README.md [hunk 2]
```diff
@@ -22,15 +22,17 @@ npm install -g @stoplight/prism-cli
...
 ```
 
-For more installation options, see [Getting Started > Installation](https://stoplight.io/p/docs/gh/stoplightio/prism/docs/getting-started/installation.md)
+For more installation options, see our [installation documentation](./docs/getting-started/01-installation.md).
 
-## Getting Started
+## Documentation
 
-After [installation](https://stoplight.io/p/docs/gh/stoplightio/prism/docs/getting-started/installation.md) take a look at our [getting started documentation](https://stoplight.io/p/docs/gh/stoplightio/prism/docs/getting-started/concepts.md).
+You can find ![Prism documentation on Stoplight.io](https://stoplight.io/p/docs/gh/stoplightio/prism).
```
Comment (XVincentX) [inline README.md]: I guess you didn't want to link it as an image

```suggestion
You can find [Prism documentation on Stoplight.io](https://stoplight.io/p/docs/gh/stoplightio/prism).
```

File: README.md [hunk 3]
```diff
@@ -22,15 +22,17 @@ npm install -g @stoplight/prism-cli
...
+
+- [Getting Started](./docs/getting-started/01-installation.md)
+- [Guides](./docs/guides/01-mocking.md)
 
 ## What's next for Prism?
 
-- [x] Server Validation
```
Comment (XVincentX) [inline README.md]: Why are you removing this?

File: README.md [hunk 4]
```diff
@@ -22,15 +22,17 @@ npm install -g @stoplight/prism-cli
...
 - [x] Content Negotiation
 - [x] Security Validation
 - [ ] Custom Mocking
```
Comment (XVincentX) [inline README.md]: Does this whole section still make sense?

File: docs/getting-started/01-installation.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+```bash
+npm install -g @stoplight/prism-cli
+# or
+yarn global add @stoplight/prism-cli
+```
+
+## Executable Binaries
+
+For users without Node and/or NPM/Yarn, we provide standalone packages for [all major platforms](https://github.com/stoplightio/prism/releases). The quickest way to install the appropriate package for your operating system is via this shell script:
```
Comment (XVincentX) [inline docs/getting-started/01-installation.md]:
```suggestion
For users without Node and/or NPM/Yarn, we provide standalone binaries for [all major platforms](https://github.com/stoplightio/prism/releases). The quickest way to install the appropriate package for your operating system is via this shell script:
```

File: docs/getting-started/01-installation.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+<!-- theme: info -->
+> The binaries do _not_ auto-update, so you will need to run it again to install new versions.
+
+## Docker
+
+Prism is available as a Docker image. We recommend specifying the major version you'd like to use as a tag:
+
+```bash
+docker run -P stoplight/prism:3 mock -h 0.0.0.0 api.oas2.yml
```
Comment (XVincentX) [inline docs/getting-started/01-installation.md]: It is important use the `--init` flat to inject the `tini` process manager, otherwise the SIGINT command won't be handled correctly — and it's important for CLI images

```suggestion
docker run --init -P stoplight/prism:3 mock -h 0.0.0.0 api.oas2.yml
```

File: docs/getting-started/01-installation.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+```
+
+If the document you want to lint is on your computer, you'll need to mount the directory where the file resides as a volume:
+
+```bash
+docker run --rm -it -v $(pwd):/tmp stoplight/spectral lint "/tmp/file.yaml"
```
Comment (XVincentX) [inline docs/getting-started/01-installation.md]:
```suggestion
docker run --init --rm -it -v $(pwd):/tmp -P stoplight/prism:3 mock -h 0.0.0.0 "/tmp/file.yaml"
```

File: docs/getting-started/01-installation.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+## Docker
+
+Prism is available as a Docker image. We recommend specifying the major version you'd like to use as a tag:
+
+```bash
+docker run -P stoplight/prism:3 mock -h 0.0.0.0 api.oas2.yml
+```
+
+If the document you want to lint is on your computer, you'll need to mount the directory where the file resides as a volume:
```
Comment (XVincentX) [inline docs/getting-started/01-installation.md]:
```suggestion
If the document you want to mock is on your computer, you'll need to mount the directory where the file resides as a volume:
```

File: docs/getting-started/03-cli.md [hunk 9]
```diff
@@ -15,9 +15,11 @@ prism mock https://raw.githack.com/OAI/OpenAPI-Specification/master/examples/v3.
 ●  note      DELETE     http://127.0.0.1:4010/pets/10
 ```
 
-The generated paths will have their parameters (query or path) bolded and in color.
+Here you can see all the "operations" (ak.k.a endpoints or resources) that Prism has found in your
```
Comment (XVincentX) [inline docs/getting-started/03-cli.md]:
```suggestion
Here you can see all the "operations" (a.k.a endpoints or resources) that Prism has found in your 
```

File: docs/getting-started/03-cli.md [hunk 10]
```diff
@@ -84,24 +86,25 @@ curl -v http://127.0.0.1:4010/pets/123?__dynamic=false
 
 This command creates an HTTP server that will proxy all the requests to the specified upstream server. Prism will analyze the request coming in and the response coming back from the upstream server and report the discrepancies with what's declared in the provided OpenAPI document.
 
-**Note:** Currently the proxy command _expects_ a JSON Payload for both the request and the response. If the response's `content-type` header does not indicate a JSON-ish payload (`application/json`, `application/*+json`), then the payload will be just returned as text and send back to the original client.
+Learn more about the ideas here in our [Validation Proxy guide](../guides/03-validation-proxy.md), or see below for the quick n dirty how to CLI.
 
-```
-prism proxy examples/petstore.oas2.yaml https://petstore.swagger.io/v2
+```bash
+$ prism proxy examples/petstore.oas2.yaml https://petstore.swagger.io/v2
```
Comment (XVincentX) [inline docs/getting-started/03-cli.md]: Adding the dollar sign will be a problem for the users interested in copying and pasting these command directly in their terminals. I suggest we remote it.

```suggestion
prism proxy examples/petstore.oas2.yaml https://petstore.swagger.io/v2
```

File: docs/guides/multiple-documents.md [hunk 11]
```diff
@@ -1,7 +1,6 @@
-# Serving multiple OAS documents
+# Serving Multiple OpenAPI Documents
 
-A single Prism instance serves one OAS document.
-In order to serve multiple documents, the best way is to usually have multiple instances of Prism running on different documents and then serve their content using a reverse Proxy. We're going to show an example using multiple docker containers.
+A single Prism instance serves one OpenAPI document, but serving multiple documents is possible by running Prism on different documents, then serving their content using a reverse Proxy.
```
Comment (XVincentX) [inline docs/guides/multiple-documents.md]:
```suggestion
A single Prism instance serves one OpenAPI document, but serving multiple documents is possible by running multiple Prism instances on different documents, then serving their content using a reverse Proxy. 
```

File: README.md (hunk 3 above)
Comment (philsturgeon) [reply README.md]: It’s old news. We don’t need to keep everything in here forever. This one is less exciting or newsworthy, but the others can go soon too.

File: README.md (hunk 4 above)
Comment (philsturgeon) [reply README.md]: When we get better at blogging, documenting, and announcing new features, this can go away. For now it’s handy to catch eyes, and maybe get people interested in contributing. We could link to relevant issues to make them even more likely to contribute.

File: .github/workflows/automerge.yml (hunk 1 above)
Comment (philsturgeon) [reply .github/workflows/automerge.yml]: Copied from Spectral, this has nothing to do with dependabot. It’s handy to add when you’re waiting for a branch to update and you’re expecting (or already have) approval. You can add this label and wander off to do something more useful.

Comment (XVincentX) [reply .github/workflows/automerge.yml]: > this has nothing to do with dependabot

Why did you remove the auto-approve for dependabot then?

Apart from this anyway, I do not really see the value at the moment — this is not a bottleneck in this project. Can we remove/reevaluate?

File: docs/guides/03-validation-proxy.md [hunk 12]
```diff
@@ -0,0 +1,45 @@
+# Validation Proxy
+
+Unlike mocking, the validation proxy expects there to be a real API running somewhere. This could be a localhost environment, docker container, hosted sandbox environment, or deployed to actual production, doesn't matter.
```
Comment (XVincentX) [inline docs/guides/03-validation-proxy.md]: Is this correct?

> the validation proxy expects there to be a real API running somewhere

File: docs/guides/03-validation-proxy.md [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+
+The proxy feature will help you identify discrepancies between the OpenAPI document (supposedly the source of truth) and any other server you designate as the proxy. This can help frontend developers integrating with your API, or other backend developers, who might want to channel their requests through Prism to see if they are making valid requests. The proxy can also be enabled in staging or any other pre-production environment, like a dress rehearsal before the opening night of a play.
```
Comment (XVincentX) [inline docs/guides/03-validation-proxy.md]: I'd say the server is not the proxy, it's the target

```suggestion
The proxy feature will help you identify discrepancies between the OpenAPI document (supposedly the source of truth) and any other server you designate as the target. This can help frontend developers integrating with your API, or other backend developers, who might want to channel their requests through Prism to see if they are making valid requests. The proxy can also be enabled in staging or any other pre-production environment, like a dress rehearsal before the opening night of a play.
```

File: docs/guides/03-validation-proxy.md (hunk 12 above)
Comment (philsturgeon) [reply docs/guides/03-validation-proxy.md]: Yes that is what the upstream server is?

Comment (XVincentX) [reply docs/guides/03-validation-proxy.md]: >  there to be 

I didn't expect such locution to be correct, I've learned a new thing :)

File: .github/workflows/automerge.yml (hunk 1 above)
Comment (philsturgeon) [reply .github/workflows/automerge.yml]: OH I had no idea I removed the dependabot. I saw no .github/workflows folder, so I coped over the one from Spectral and considered it a happy accident that we got `automerge` label support. We started it on Spectral with the goal of moving it over, but if your PR cadence is low enough that it doesn't need to happen then fantastic! I'll put things back as they were.

File: docs/getting-started/01-installation.md (hunk 7 above)
Comment (philsturgeon) [reply docs/getting-started/01-installation.md]: I'm glad you're noticing all these issues because this documentation has been in master for ages. I'm just moving existing content around. 😬

File: docs/getting-started/03-cli.md (hunk 10 above)
Comment (philsturgeon) [reply docs/getting-started/03-cli.md]: Common convention is to use the dollar sign to distinguish commands from output when both are mixed into a single code fence. The alternative is breaking them into two code fences, which is a bit less common back can look nice.

File: docs/getting-started/01-installation.md (hunk 7 above)
Comment (XVincentX) [reply docs/getting-started/01-installation.md]: I'm afraid it's incorrect — the current documentation is fine — I guess this is a leftover from Spectral's copy paste:

https://github.com/stoplightio/prism/blob/master/docs/getting-started/installation.md#docker

**Comment by philsturgeon** [comment]
@XVincentX alright, got it!
